### PR TITLE
Fix #86

### LINKED ISSUE #86: Trace validation gives exception
Using the "trace" element in a .lc file gives this exception in the log:
````
eclipse.buildId=4.31.0.20240307-1200
java.version=17.0.10
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=de_DE
Framework arguments:  -product org.eclipse.epp.package.committers.product
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.eclipse.epp.package.committers.product

org.apache.log4j
Error
Sun Sep 01 09:33:52 CEST 2024
org.eclipse.xtext.validation.CompositeEValidator  - Error executing EValidator

java.lang.NoSuchMethodError: 'java.util.Hashtable org.eclipse.pde.internal.core.TracingOptionsManager.getTemplateTable(java.lang.String)'
	at com.wamas.ide.launching.validation.LcDslValidator.checkTracingOptions(LcDslValidator.java:705)
	at jdk.internal.reflect.GeneratedMethodAccessor136.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.xtext.validation.AbstractDeclarativeValidator$MethodWrapper.invoke(AbstractDeclarativeValidator.java:129)
	at org.eclipse.xtext.validation.AbstractDeclarativeValidator.internalValidate(AbstractDeclarativeValidator.java:337)
	at org.eclipse.xtext.validation.AbstractInjectableValidator.validate(AbstractInjectableValidator.java:72)
	at org.eclipse.xtext.validation.CompositeEValidator.validate(CompositeEValidator.java:151)
	at org.eclipse.emf.ecore.util.Diagnostician.doValidate(Diagnostician.java:299)
	at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:245)
	at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:200)
	at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:130)
	at com.wamas.ide.launching.generator.StandaloneLaunchConfigGenerator.hasError(StandaloneLaunchConfigGenerator.java:299)
	at com.wamas.ide.launching.generator.StandaloneLaunchConfigGenerator.generate(StandaloneLaunchConfigGenerator.java:134)
	at com.wamas.ide.launching.generator.LcDslGenerator.doGenerate(LcDslGenerator.java:44)
	at org.eclipse.xtext.generator.GeneratorDelegate.doGenerate(GeneratorDelegate.java:44)
	at org.eclipse.xtext.generator.GeneratorDelegate.generate(GeneratorDelegate.java:35)
	at org.eclipse.xtext.builder.BuilderParticipant.handleChangedContents(BuilderParticipant.java:597)
	at org.eclipse.xtext.builder.BuilderParticipant.handleChangedContents(BuilderParticipant.java:578)
	at org.eclipse.xtext.builder.BuilderParticipant.doGenerate(BuilderParticipant.java:563)
	at org.eclipse.xtext.builder.BuilderParticipant.doBuild(BuilderParticipant.java:303)
	at org.eclipse.xtext.builder.BuilderParticipant.build(BuilderParticipant.java:265)
	at org.eclipse.xtext.builder.impl.RegistryBuilderParticipant$DeferredBuilderParticipant.build(RegistryBuilderParticipant.java:164)
	at org.eclipse.xtext.builder.impl.RegistryBuilderParticipant.build(RegistryBuilderParticipant.java:70)
	at org.eclipse.xtext.builder.impl.XtextBuilder.doBuild(XtextBuilder.java:526)
	at org.eclipse.xtext.builder.impl.XtextBuilder.addInfosFromTaskAndBuild(XtextBuilder.java:456)
	at org.eclipse.xtext.builder.impl.XtextBuilder.incrementalBuild(XtextBuilder.java:437)
	at org.eclipse.xtext.builder.impl.XtextBuilder.build(XtextBuilder.java:340)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:1077)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:296)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:352)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:441)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:444)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:555)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:503)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:585)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:207)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:300)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
````

The method has now a 2nd argument, see:

https://github.com/eclipse-pde/eclipse.pde/blob/master/ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/TracingOptionsManager.java#L47

### PR DESCRIPTION
**Pull request by frankbenoit**

### CODE REVIEW

File: com.wamas.ide.launching/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -15,7 +15,7 @@ Require-Bundle: org.eclipse.xtext,
  org.eclipse.xtend.lib,
  org.eclipse.emf.common,
  org.eclipse.core.resources,
- org.eclipse.pde.core;bundle-version=3.15.100,
+ org.eclipse.pde.core;bundle-version="3.18.100",
```
Comment (Bananeweizen) [inline com.wamas.ide.launching/META-INF/MANIFEST.MF:18]: 3.17.200.v20231118 should be the first one containing the underlying code change f2b9ed797620fc5cad2a25ca27d289fee5961ac9, therefore 3.17.200 might be better for backwards compatibility

**Comment by haubi** [comment]
Thanks for the PR!
However, this would stop supporting older Eclipse releases, which is... unfortunate at least, and would require a major revision bump for lcdsl.
Instead, and sorry for the inconvenience, would you mind to add support for both variants?

**Comment by frankbenoit** [comment]
you mean by using reflection? Is such used in this project elsewhere so I can take the same approach?

**Comment by haubi** [comment]
Yes, reflection it is.
However, while I remember 2 of such situations in Eclipse TEA, none of them was about a method signature change so far:
- https://github.com/eclipse-tea/tea/commit/1a0a1f6255b635e7265b5159c2d4fe8628a41a13 about a behavioural change
- https://github.com/eclipse-tea/tea/commit/6ee7cd13aa7a135b4b943809f0ccb3329fb7e44d about a class name change

**Comment by frankbenoit** [comment]
I tried the LC DSL editor with the 2023-09 and 2024-06.

**Pull request merged**